chorus_lib/src/transport.rs

```diff
@@ -2,3 +2,66 @@
 
 pub mod http;
 pub mod local;
+
+use crate::core::{ChoreographyLocation, HCons, HList};
+use crate::LocationSet;
+use std::collections::HashMap;
+use std::marker::PhantomData;
+
+/// A generic struct for configuration of `Transport`.
+#[derive(Clone)]
+pub struct TransportConfig<L: HList, InfoType, TargetLocation: ChoreographyLocation, TargetInfoType>
+{
+    /// The information about locations
+    pub info: HashMap<String, InfoType>,
+    /// The information about the target choreography
+    pub target_info: (TargetLocation, TargetInfoType),
+    /// The struct is parametrized by the location set (`L`).
+    pub location_set: PhantomData<L>,
```

Comment:
You probably don't want to `pub` this

Reply:
do I make all the fields private? It works currently

Reply:
No, we need to have other fields public so when people use it for their custom transport they can access the fields